extensions/wikia/SpecialDiscussionsLog/SpecialDiscussionsLogController.class.php

```diff
@@ -9,203 +12,306 @@ class SpecialDiscussionsLogController extends WikiaSpecialPageController {
+	 * @return array
+	 */
+	private function readAndValidateInput() {
+		$userName = $this->getVal( UserQuery::getKeyName(), null );
+		$ipAddress = $this->getVal( IpAddressQuery::getKeyName(), null );
+
+		if ( !empty( $userName ) && !empty( $ipAddress ) ) {
+			throw new \InvalidArgumentException( wfMessage( 'discussionslog-multiple-input-error' )->escaped() );
+		}
+
+		return [ $userName, $ipAddress ];
+	}
+
+	private function requestUserLog( SearchQuery $searchQuery ) {
+		$this->setVal(
+			'userLog',
+			$this->sendSelfRequest( 'userLog', [ $searchQuery->getKeyName() => $searchQuery->getKey() ] ) );
+	}
+
 	public function inputForm() {
-		$this->userName = $this->getVal( 'username' );
-		$this->userNameLabel = wfMessage( 'discussionslog-username-label' )->escaped();
-		$this->viewLogsAction = wfMessage( 'discussionslog-view-logs' )->escaped();
+		list( $userName, $ipAddress ) = $this->readAndValidateInput();
+
+		$this->response->setValues( [
+			'userName' => $userName,
+			'ipAddress' => $ipAddress,
+			'userNameLabel' => wfMessage( 'discussionslog-username-label' )->escaped(),
+			'ipAddressLabel' => wfMessage( 'discussionslog-ip-address-header' )->escaped(),
+			'orLabel' => wfMessage( 'discussionslog-or-label' )->escaped(),
+			'viewLogsAction' => wfMessage( 'discussionslog-view-logs' )->escaped(),
+		] );
 	}
 
 	private function constructKibanaUrl( $dayOffset ) {
 		global $wgConsulUrl;
 
-		$esUrl = (new  Wikia\Service\Gateway\ConsulUrlProvider( $wgConsulUrl, 'query', 'sjc' ))->getUrl( 'es' );
+		$esUrl = (new Wikia\Service\Gateway\ConsulUrlProvider( $wgConsulUrl, 'query', 'sjc' ))->getUrl( 'es' );
 		$date = time() - ( $dayOffset * 24 * 60 * 60 );
 
 		return 'http://' . $esUrl . '/logstash-' . date( 'Y.m.d', $date ) . '/_search';
 	}
 
-	private function getUserIdByUsername( $userName ) {
+	private function getUserByUsername( $userName ) {
 		if ( empty( $userName ) ) {
-			throw new InvalidArgumentException( self::NO_USER_MATCH_ERROR );
+			throw new \InvalidArgumentException( wfMessage( self::NO_USER_MATCH_ERROR )->escaped() );
 		}
 
 		$user = User::newFromName( $userName );
-		if ( !$user || empty( $user->getId() ) ) {
-			throw new InvalidArgumentException( self::NO_USER_MATCH_ERROR );
+		if ( !$user ) {
+			throw new \InvalidArgumentException( wfMessage( self::NO_USER_MATCH_ERROR )->escaped() );
+		}
+
+		return $user;
+	}
+
+	private function getUserById( $userId ) {
+		if ( !$userId ) {
+			throw new \InvalidArgumentException( wfMessage( self::NO_USER_MATCH_ERROR )->escaped() );
+		}
+
+		// Look at cache first
+		if ( !empty( $this->users[ $userId ] ) ) {
+			return $this->users[$userId];
+		}
+
+		$user = User::newFromId( $userId );
+		if ( !$user ) {
+			throw new \InvalidArgumentException( wfMessage( self::NO_USER_MATCH_ERROR )->escaped() );
 		}
 
-		return $user->getId();
+		$this->users[ $user->getId() ] = $user;
+
+		return $user;
 	}
 
 	public function userLog() {
```

Comment:
I realized that having this as public is actually a security concern since it's available via our API. I was able to access this while logged out by going to `http://jsutterfield.james.wikia-dev.com/wikia.php?controller=SpecialDiscussionsLog&method=userLog&username=james` and see the results.

2 ways I can think to fix this. Either we put a permissions check at the start of this function, or move the logic to get the userLog info into a separate private method and only have this responsible for passing those values to the template.